File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for (int i = 0 ; i < n ; i++){
            if (!visited[i]){
                dfs(computers, visited, i, n);
                answer++;
            }
        }
        return answer;
    }
    
    private void dfs(int[][] computers, boolean[] visited, int node, int l){
        visited[node] = true;
        
        for(int i = 0 ; i < l ; i++){
            if(computers[node][i] == 1 && !visited[i]){
                dfs(computers, visited, i, l);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    
    const visited = Array(n).fill(false);
    
    const find_network = (s) => {
        visited[s] = true;
        
        let head = 0;
        const q = [s];
        
        while(head < q.length){
            const now = q[head++];
            
            for(let next=0; next<n; next++){
                if(computers[now][next] === 0) continue;
                if(visited[next] === true) continue;
                q.push(next);
                visited[next] = true;
            }
        }
    }
    
    for(let i=0; i<n; i++){
        if(visited[i] === true) continue;
        find_network(i);
        answer++;
    }
    
    return answer;
}